File: 2864-maximum-odd-binary-number/2864-maximum-odd-binary-number.java
class Solution {
    public String maximumOddBinaryNumber(String s) {
        int n = s.length();
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < n; i++) {
            result.append('0');
        }
        int i = 0;
        for (char ch : s.toCharArray()) {
            if (ch == '1') {
                if (result.charAt(n-1) == '1') {
                    result.setCharAt(i, '1');
                    i++;
                } else {
                    result.setCharAt(n - 1, '1');
                }
            }
        }
        return result.toString();
    }
}

File: 2864-maximum-odd-binary-number/2864-maximum-odd-binary-number.py
class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        n = len(s)
        result = "0" * n
        i = 0
        for ch in s:
            if ch == '1':
                if result[n - 1] == '1':
                    result = result[:i] + '1' + result[i+1:]
                    i += 1
                else:
                    result = result[:n-1] + '1'
        return result